commit adc493af8cc5b74fab02190f2862a4af19fae996
Author: Matthew Booth <mbooth@redhat.com>
Date:   Mon Aug 20 10:41:02 2012 +0100

    rhev: Ensure vm_snapshot_id is unique
    
    Reported and diagnosed by Andrey Gordeev:
    -----
    Got error "Failed to import Vm <vmName> to <storageName>" after
    running virt-v2v, also found error in engine.log:
    
    2012-08-16 16:39:30,090 ERROR
    [org.ovirt.engine.core.bll.ImportVmCommand] (pool-3-thread-50)
    [2781049c] Command
    org.ovirt.engine.core.bll.ImportVmCommand throw exception:
    org.springframework.dao.DataIntegrityViolationException:
    CallableStatementCallback; SQL [{call insertsnapshot(?, ?, ?, ?, ?, ?,
    ?, ?)}]; ERROR: duplicate key value violates
    unique constraint "pk_snapshots"
    
    Where: SQL statement "INSERT INTO snapshots( snapshot_id, status,
    vm_id, snapshot_type, description, creation_date,
    app_list, vm_configuration) VALUES(  $1 ,  $2 ,  $3 ,  $4 ,  $5 ,  $6
    ,  $7 ,  $8 )"
    
    I discovered that this happened because i had 2 disks and virt-v2v set
    ovf:vm_snapshot_id=“00000000-0000-0000-0000-000000000000” to both
    disks.
    When I replaced this ids to unique ids generated by uuid, then vm was
    succesfully imported.
    -----
    (cherry picked from commit b41ff38efcf8525d2b335a4870607eb40d78c6c4)

diff --git a/lib/Sys/VirtConvert/Connection/RHEVTarget.pm b/lib/Sys/VirtConvert/Connection/RHEVTarget.pm
index 2b6e9e8..ad168f0 100644
--- a/lib/Sys/VirtConvert/Connection/RHEVTarget.pm
+++ b/lib/Sys/VirtConvert/Connection/RHEVTarget.pm
@@ -1039,8 +1039,7 @@ sub _disks
         $diske->setAttribute('ovf:actual_size', $usage_gb);
         $diske->setAttribute('ovf:fileRef', $fileref);
         $diske->setAttribute('ovf:parentRef', '');
-        $diske->setAttribute('ovf:vm_snapshot_id',
-                             '00000000-0000-0000-0000-000000000000');
+        $diske->setAttribute('ovf:vm_snapshot_id', get_uuid());
         $diske->setAttribute('ovf:volume-format', $vol->_get_rhev_format());
         $diske->setAttribute('ovf:volume-type', $vol->_get_rhev_type());
         $diske->setAttribute('ovf:format', 'http://en.wikipedia.org/wiki/Byte');
